search/populate.py

```diff
@@ -0,0 +1,52 @@
+from io import BytesIO
+
+from flask import Blueprint, Response
+from flask import current_app as app
+
+import requests
+from elasticsearch import Elasticsearch
+from lxml import etree
+
+from search.utils import strip_slashes
+
+
+def get_populate_blueprint() -> Blueprint:
+    blueprint = Blueprint('populate', __name__)
+
+    @blueprint.route('/populate', methods=['POST'])
+    def populate() -> Response:
+        """
+        Route for triggering indexing of known paths. Will be used as a proof of
+        concept. This should be removed before MVP release.
+        """
+        search = Elasticsearch(hosts=app.config['ELASTICSEARCH_HOSTS'])
+        gateway = strip_slashes(app.config['GATEWAY_URL'])
+
+        for path in app.config['PATHS_TO_INDEX']:
+            url = f'{gateway}/{strip_slashes(path)}'
+            response = requests.get(url)
+            response.raise_for_status()
+            content = response.content
+            root = etree.parse(BytesIO(content)).getroot()
+            for item in root.iter('{http://libero.pub}item-ref'):
+                item_url = url + f'/{item.attrib.get("id")}/versions/latest'
+                response = requests.get(item_url)
+                response.raise_for_status()
+                content = response.content
+                tree = etree.parse(BytesIO(content))
```

Comment:
Seems a candidate for a private function for this module:
```
def choose_a_name(content):
    return etree.parse(BytesIO(content))
```

Reply:
I don't see the value in that.

Reply:
It's just duplicated code, in this case just a duplicated line across a single function so won't bite

Reply:
The code above is outdated. Please see the latest changes.